tests/unit/EmojiTest.ts

```diff
@@ -154,6 +154,11 @@ describe('EmojiTest', () => {
 
     it('correct suggests emojis accounting for keywords', () => {
         const thumbEmojisEn: Emoji[] = [
+            {
```

Comment:
Why did we move this back up? We moved it down for this issue: https://github.com/Expensify/App/pull/48748

Reply:
@srikarparsi After we sort the suggestion emoji, the position of the emojis in the test is changed [here](https://github.com/Expensify/App/actions/runs/11453388340/job/31865783084)

Reply:
Hm, when I test it manually, the wrong emoji comes up first
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/1c4480d0-18f7-43d1-94b6-feb04fc5d161">

Reply:
@srikarparsi I think that's expected since we're prioritizing sorting by emoji name [here](https://github.com/Expensify/App/pull/51220/files#diff-1692fc6ebb559e5fd701e993d1e71ebee2b6ed378c84d2a04654b3ff3ff20c50R27).

Reply:
Hm, I don't think that's expected. Based on this [issue](https://github.com/Expensify/App/pull/48748), I believe we want `:thumb` to show the thumbs up emoji first instead of the hand_with_index_finger emoji in the picture.

Reply:
@suneox what do you think in this case?

Reply:
I think we can replace the [thumbsup](https://github.com/nkdengineer/App/blob/c0a7a2f31e8b5c5a943720f9cfb045c18a876121/assets/emojis/common.ts#L833) icon name and add [+1](https://github.com/nkdengineer/App/blob/c0a7a2f31e8b5c5a943720f9cfb045c18a876121/assets/emojis/en.ts#L582) to the list of keywords. This approach maintains the sorting priority by emoji name without causing [this issue](https://github.com/Expensify/App/pull/48748).

Alternatively, we could update the search logic to function similarly to Slack, where it searches by both icon name and keyword. This would make it more flexible and intuitive for users.

https://github.com/user-attachments/assets/11f03d9f-06ac-4951-a6fd-599f4a449798

Reply:
> Alternatively, we could update the search logic to function similarly to Slack, where it searches by both icon name and keyword. This would make it more flexible and intuitive for users.

@suneox please explain in more detail with this solution

Reply:
> @suneox please explain in more detail with this solution

That means this is a new solution where the emoji name will display based on a matched keyword instead of a fixed name

https://github.com/user-attachments/assets/11f03d9f-06ac-4951-a6fd-599f4a449798

Reply:
I'll give an update soon

Reply:
> That means this is a new solution where the emoji name will display based on a matched keyword instead of a fixed name

I'm still confused here. If we display based on a matched keyword instead of a fixed name, in many cases it will not show results that more closely match the text and conflict with expected in OP

Reply:
@srikarparsi WDYT about changing the filter logic at this time, or should we just adjust the emoji name in [this comment](https://github.com/Expensify/App/pull/51220#discussion_r1832400796)?

> I think we can replace the [thumbsup](https://github.com/nkdengineer/App/blob/c0a7a2f31e8b5c5a943720f9cfb045c18a876121/assets/emojis/common.ts#L833) icon name and add [+1](https://github.com/nkdengineer/App/blob/c0a7a2f31e8b5c5a943720f9cfb045c18a876121/assets/emojis/en.ts#L582) to the list of keywords. This approach maintains the sorting priority by emoji name without causing https://github.com/Expensify/App/pull/48748.

Reply:
@srikarparsi can you explain a little more why we would do that? I actually considered doing that for the PR you linked but decided against it because those are the "real" names [according to unicode](https://unicode.org/emoji/charts/full-emoji-list.html).

I would heavily advise against changing any of the names

Reply:
Yeah, so the sorting function that @flodnv suggested was to `prioritize those that more closely match the text`, text I think means the name of the emoji.

<img width="299" alt="image" src="https://github.com/user-attachments/assets/60c94d07-7e1e-4b1b-ad5b-f3ad9c764249">

So the [sorting function](https://github.com/Expensify/App/pull/51220/files#diff-1692fc6ebb559e5fd701e993d1e71ebee2b6ed378c84d2a04654b3ff3ff20c50R27) in this PR basically sorts by whether or not the name includes the input text (so that `disappointed` would come up before `melting_face` for `:dis`.

But this would mean that `hand_with_index_finger_and_thumb_crossed` comes before `+1` for `:thu` since `+1` doesn't include `:thu`. I think in the future, we would have a combination of frequency, name and keywords for suggestions (I think slack does that and included screenshots above). But I think this is a good first step, it just doesn't work for the thumbs up/down case because of the name. So, I was thinking if we want `:thu` to generate `+1` as the first option then we could switch the keyword with the name. The other option would be to do nothing and people would either have to scroll one down or type `+1` until we develop a more nuanced algorithm with emoji frequencies.

Reply:
> FWIW, :thumbsup: is the emoji name on slack.
I'm not sure if it's the "name" on slack, I think slack just shows the keyword for an emoji when the input text is closer to the keyword. I added examples for `:thu` in this [comment](https://github.com/Expensify/App/pull/51220#discussion_r1846180526).

> Can we just rename it to that?
I am leaning towards this and also including a comment in the code to change it back to `+1` when we prioritize creating a better sorting algorithm that intakes frequency, keywords and name.

> Or have both names?
I don't think this is possible with the current structure, [Emoji name](https://github.com/Expensify/App/blob/763215e0aa0ac256f5f129e7f62444a466e84e8f/assets/emojis/types.ts#L3-L7) is a string

Reply:
My vote would be either to do nothing, or update the sorting algorithm now to include keywords since "thumbsup" is already in that array.
<img width="803" alt="2024-11-19_14-02-07" src="https://github.com/user-attachments/assets/fdde64ff-8d8c-4a13-900d-a0be385ec1df">

Reply:
> FWIW, `:thumbsup:` is the emoji name on slack

The name in slack is `:+1:` 😄 Even if you type `:thumbsup:`, when you highlight it, it will always say `+1`

<img width="82" alt="2024-11-19_14-03-05" src="https://github.com/user-attachments/assets/aac416eb-95cb-42b3-adb6-1b2a23ef456e">

Reply:
Hm okay, updating the sorting algorithm to include keywords wouldn't work for this example right? Because `melting_face` 🫠  has keywords that start with `dis` even though the name doesn't.
https://private-user-images.githubusercontent.com/48188732/387438434-60c94d07-7e1e-4b1b-ad5b-f3ad9c764249.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNDMzMjIsIm5iZiI6MTczMjA0MzAyMiwicGF0aCI6Ii80ODE4ODczMi8zODc0Mzg0MzQtNjBjOTRkMDctN2UxZS00YjFiLWFkNWItZjNhZDljNzY0MjQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE5VDE5MDM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyOTE5Mjg4N2Q1OGUyMTIyMzNiODdjY2UyOTRmMDQzM2FiYjY2NGNhMjlhYmM5ZTFlZTUzNjkwZDRkOWNjODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cbzaBsUPToI6olY75SBFKFDk4lJEpnYmh7bkWcXpIag

I am leaning towards changing the names of +1 and -1 to thumbsup and thumbsdown and including a comment explaining to change the name back once we incorporate frequency into our sorting algorithm. Does that sound okay with you?

If not, I am also fine with doing nothing for now for this edge case.

Reply:
Yeah let's do nothing. Keeping the same names as unicode seems more important than this one edge case IMO

Reply:
Soundss good, going to merge this one then